### PR TITLE
React: add onToggle handler to <details />

### PR DESCRIPTION
**Pull request by alpynv**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present). (fails, but not because of my changes, confused 🤔)

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
Docs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#Events
React issue: https://github.com/facebook/react/issues/8761
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/react/index.d.ts [hunk 1]
```diff
@@ -1979,6 +1979,7 @@ declare namespace React {
 
     interface DetailsHTMLAttributes<T> extends HTMLAttributes<T> {
         open?: boolean;
+        onToggle?: MouseEventHandler<T>;
```
Comment (eps1lon) [inline types/react/index.d.ts:1982]: According to MDN [it should be a generic event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLDetailsElement/toggle_event).

```suggestion
        onToggle?: ReactEventHandler<T>;
```

Do you have some docs reference that this indeed implements React's MouseEvent?

**Review by eps1lon — CHANGES_REQUESTED** [review]

File: types/react/index.d.ts (hunk 1 above)
Comment (alexpyzhianov) [reply types/react/index.d.ts:1982]: Nope, you are absolutely right, it shouldn't be a mouse event, my bad

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## plotly.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/plotly.js@latest/))
was missing the following properties:
1. version
1. register
1. Icons
1. Fx
1. Snapshot

<details>
<summary>as well as these 2 other properties...</summary>
<p>PlotSchema, Queue</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 381a2a9d17d14e111f7d002c866debfcac6724e8
</p>

**Review by eps1lon — APPROVED** [review]

**Review by ferdaber — COMMENTED** [review]
Do you know if the `onToggle` event bubbles? If so, it should be placed inside `DOMAttributes` instead of `DetailsHTMLAttributes`.

**Comment by alexpyzhianov** [comment]
@ferdaber Looks like it doesn't bubble https://developer.mozilla.org/en-US/docs/Web/API/HTMLDetailsElement/toggle_event

**Review by ferdaber — APPROVED** [review]

**Comment by alexpyzhianov** [comment]
Ready to merge

**Pull request merged**